### PR TITLE
Extended TwoBitFile and NucleotideContigFragmentRDDFunctions to behave more similar

### PR DESCRIPTION
**Pull request by akmorrow13**
* Extended NucleotideContigFragmentRDDFunctions to ReferenceFile
* Added getSequenceDictionary function to TwoBitFile

### CODE REVIEW

**Comment by fnothaft** [comment]
@akmorrow13 let's chat about this sometime. `NucleotideContigFragmentRDDFunctions` goes away in #1051 . The possibly misnamed [ReferenceContigMap](https://github.com/bigdatagenomics/adam/blob/b29d20480001be20e0be55e828a147b5f222ce42/adam-core/src/main/scala/org/bdgenomics/adam/util/ReferenceContigMap.scala) class may be sufficient for your needs.

File: adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala [hunk 1]
```diff
@@ -92,6 +92,16 @@ class TwoBitFile(byteAccess: ByteAccess) extends ReferenceFile {
     name -> contigOffset
   }
 
+  def getSequenceDictionary(performLexSort: Boolean = false): SequenceDictionary = {
+    val sd = new SequenceDictionary(seqRecords.toVector.map(r => SequenceRecord(r._1, r._2.dnaSize)))
+
+    if (performLexSort) {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala]: Why's this necessary?

Comment (akmorrow13) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala]: I need the sequence dictionary from 2 bit file. Is there a better way to get it?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala]: Oh, yeah, that makes sense. My comment was with regards to the lexicographic sort.

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala]: IIRC, we've added lex sort functions to the `SequenceDictionary`, so I'd rather not dupe the code.

**Pull request closed without merging**